config.yaml

```diff
@@ -92,6 +92,10 @@ options:
     description: Config options to add extra-sans to the ones used when requesting server certificates. The extra-sans are specified by comma-separated names to be added when requesting signed certificates. Use "{unit}" as a placeholder to be filled with the unit number, e.g. "worker-{unit}" will be translated as "worker-0" for unit 0 and "worker-1" for unit 1 when requesting the certificate.
     type: string
     default: ""
+  extra_listeners:
+    description: "Config options to add extra SANs to the ones used when requesting server certificates, and to define custom `advertised.listeners` and ports for clients external to the Juju model. These items are comma-separated. Use '{unit}' as a placeholder to be filled with the unit number if necessary. For port allocations, providing the port for a given listener will offset the generated port number by that amount, with an accepted value range of 20001-50000. For example, a provided value of 'worker-{unit}.domain.com:30000' will generate listeners for unit 0 with name 'worker-0.domain.com', and be allocated ports 39092, 39093 etc for each authentication scheme."
```

Comment:
@grobbie - This config option is fairly challenging to summarize succinctly in to a useful description. We need to communicate:

- That this is used for both SANs and for listeners
- That you need to comma-delimit them
- That you can use `{unit}` to replace it with an id
- That `{unit}` is optional
- That you MUST append a port to each one, even if SANs don't need them
- That the port is only a 'base port', not the one that's actually generated
- That multiple ports will be generated for different auth schemes
- That there is a port range that we accept

If you're fine with what's presented here already, we'll stick with it. Just wanted to raise that to me, it smells like it's one-config-to-rule-them-all and probably would be more understandable broken up.
Let me know your thoughts!

Reply:
From the description I see that the increment is "1" (39092, 39093, etc) whereas from the [code](https://github.com/canonical/kafka-operator/pull/269/files#diff-a24e7472936c291b5c95f9c56ccc25cccef3adf4c4cbd1de135e2dde31ae592cR131) the increments appears to be "100" ( 39092, 39192, etc). Which one is the correct one?

Also, I would have felt that providing the first port is more intuitive, without the user working out the math

Reply:
Hey please make it 1 rather than 100. Sorry for any confusion

Reply:
Agreed, was incorrect. Addressed [`1e4bb90` (#269)](https://github.com/canonical/kafka-operator/pull/269/commits/1e4bb90926bda3d68479a23ee1e0edf845a27625)